pkg/codegen/internal/test/testdata/aws-eks.pp.ts

```diff
@@ -0,0 +1,184 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+export = async () => {
+    // VPC
+    const eksVpc = new aws.ec2.Vpc("eksVpc", {
+        cidrBlock: "10.100.0.0/16",
+        instanceTenancy: "default",
+        enableDnsHostnames: true,
+        enableDnsSupport: true,
+        tags: {
+            Name: "pulumi-eks-vpc",
+        },
+    });
+    const eksIgw = new aws.ec2.InternetGateway("eksIgw", {
+        vpcId: eksVpc.id,
+        tags: {
+            Name: "pulumi-vpc-ig",
+        },
+    });
+    const eksRouteTable = new aws.ec2.RouteTable("eksRouteTable", {
+        vpcId: eksVpc.id,
+        routes: [{
+            cidrBlock: "0.0.0.0/0",
+            gatewayId: eksIgw.id,
+        }],
+        tags: {
+            Name: "pulumi-vpc-rt",
+        },
+    });
+    // Subnets, one for each AZ in a region
+    const zones = aws.getAvailabilityZones({});
+    const vpcSubnet: aws.ec2.Subnet[];
+    for (const range of (await zones.then(zones => zones.names)).map((k, v) => {key: k, value: v})) {
+        vpcSubnet.push(new aws.ec2.Subnet(`vpcSubnet-${range.key}`, {
+            assignIpv6AddressOnCreation: false,
+            vpcId: eksVpc.id,
+            mapPublicIpOnLaunch: true,
+            cidrBlock: `10.100.${range.key}.0/24`,
+            availabilityZone: range.value,
+            tags: {
+                Name: `pulumi-sn-${range.value}`,
+            },
+        }));
+    }
+    const rta: aws.ec2.RouteTableAssociation[];
+    for (const range of (await zones.then(zones => zones.names)).map((k, v) => {key: k, value: v})) {
+        rta.push(new aws.ec2.RouteTableAssociation(`rta-${range.key}`, {
+            routeTableId: eksRouteTable.id,
+            subnetId: vpcSubnet.then(vpcSubnet => vpcSubnet[range.key].id),
+        }));
+    }
+    const subnetIds = pulumi.all([pulumi.output(vpcSubnet), __item.id]).apply(([vpcSubnet, id]) => vpcSubnet.map(__item => id));
```

Comment:
Any reason `vpcSubnet` needs to be wrapped in `pulumi.output`?

Reply:
Hm, it doesn't, does it. Good catch--can you file a codegen issue for this? Something like "do not wrap non-outputs in `pulumi.output` when calling `pulumi.all`".